Спринт 13 дз

VladislavAlbertovich opened the pull request.

VladislavAlbertovich committed:
Спринт 13 дз

i-masloed commented on app/src/main/java/com/example/playlistmarket/PlayerActivity.kt:
```diff
@@ -0,0 +1,80 @@
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_player)
+        val buttonBack = findViewById<ImageButton>(R.id.buttonBack)
+        albumImage = findViewById(R.id.album_imageView)
```
Рекомендация: удобно использовать viewBinding - https://developer.android.com/topic/libraries/view-binding

i-masloed commented on app/src/main/java/com/example/playlistmarket/PlayerActivity.kt:
```diff
@@ -0,0 +1,80 @@
+        bind()
+
+        buttonBack.setOnClickListener {
+            val intent = Intent(this, SearchActivity::class.java)
```
Критично: здесь должен быть возврат на предыдущую активити, те достаточно просто вызвать finish()
(https://developer.android.com/guide/components/activities/tasks-and-back-stack)

i-masloed commented on app/src/main/java/com/example/playlistmarket/SearchActivity.kt:
```diff
@@ -98,14 +103,14 @@ class SearchActivity : AppCompatActivity() {
             override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                 searchHistoryViewGroup.visibility =
-                    if (inputSearchEditText.hasFocus() && s?.isEmpty() == true && searchHistory.getTracksFromSharedPreferences().size > 0) View.VISIBLE else View.GONE
+                    if (inputSearchEditText.hasFocus() && s?.isEmpty() == true && getTracksFromHistory().size > 0) View.VISIBLE else View.GONE
```
Рекомендация: этот код дублируется несколько раз, имеет смысл вынести его в отдельный метод

i-masloed commented on app/src/main/java/com/example/playlistmarket/Track.kt:
```diff
@@ -2,4 +2,4 @@ package com.example.playlistmarket
 
-data class Track(val trackName: String, val artistName: String, @SerializedName("trackTimeMillis") val trackTime: Long, val artworkUrl100: String, val trackId:Int)
\ No newline at end of file
+data class Track(val trackName: String, val artistName: String, @SerializedName("trackTimeMillis") val trackTime: Long, val artworkUrl100: String, val trackId:Int, val collectionName: String, val releaseDate: String, val primaryGenreName: String, val country: String )
```
- В java и котлин принят определенный стиль форматирования кода. Если не вдаваться в подробности, то отформатировать код можно быстрой комбинаций клавиш Ctrl+Alt+L(Windows) или  (⌘+⌥+L)(Mac)

VladislavAlbertovich committed:
Спринт 13 дз
Исправлено: -отображение даты в PlayerActivity
-Размер обложки стал изменяем в зависимости от размера экрана
-При нажатии назад удалил создание новой активити, вместо этого вставлен метод finish()
-Создан отдельный метод для изменения видимости searchHistoryViewGroup
-Изменен стиль форматирование в классе Track
-Добавлен файл кнопки add_to_playlist.xml
-В классе PlayerActivity вместо findViewById теперь используется viewBinding
